packages/client/src/object/convertWireToOsdkObjects/createOsdkObject.ts

```diff
@@ -140,50 +76,65 @@ export function createOsdkObject<
   objectDef: Q,
   rawObj: OntologyObjectV2,
 ): Osdk<ObjectTypeDefinition, any> {
-  // We use multiple layers of prototypes to maximize reuse and also to keep
-  // [RawObject] out of `ownKeys`. This keeps the code in the proxy below simpler.
-  const objectHolderPrototype = Object.create(
-    objectPrototypeCache.get(client, objectDef),
-    {
-      [RawObject]: {
-        value: rawObj,
-        writable: true, // so we can allow updates
-      },
-      [CachedPropertiesRef]: {
-        value: new Map<string | symbol, any>(),
-        writable: true,
-      },
+  // updates the object's "hidden class/map".
+  Object.defineProperties(rawObj, {
+    [ObjectDefRef]: { value: objectDef, enumerable: false },
+    [ClientRef]: { value: client, enumerable: false },
+    [RawObject]: {
+      value: rawObj,
+      enumerable: false,
     },
-  );
-
-  // we separate the holder out so we can update
-  // the underlying data without having to return a new object
-  // we also need the holder so we can customize the console.log output
-  const holder: ObjectHolder<Q> = Object.create(objectHolderPrototype);
+    ...basePropDefs,
 
-  const osdkObject: any = new Proxy(holder, {
-    ownKeys(target) {
-      return Reflect.ownKeys(target[RawObject]);
+    [UnderlyingOsdkObject]: {
+      enumerable: false,
+      value: rawObj,
     },
-    get(target, p, receiver) {
-      switch (p) {
-        case UnderlyingOsdkObject:
-          // effectively point back to the proxy
-          return receiver;
-      }
+  });
+
+  // Assign the special values
+  for (const propKey of Object.keys(rawObj)) {
+    if (
+      propKey in objectDef.properties
+      && specialPropertyTypes.has(objectDef.properties[propKey].type)
+    ) {
+      const rawValue = rawObj[propKey as any];
+      const propDef = objectDef.properties[propKey as any];
+
+      rawObj[propKey] = createSpecialProperty(
+        client,
+        objectDef,
+        rawObj as any,
+        propKey,
+      );
+    }
+  }
 
-      if (p in target) return target[p as keyof typeof target];
+  return Object.freeze(rawObj) as Osdk<ObjectTypeDefinition, any>;
+}
 
-      if (p in rawObj) {
-        const rawValue = target[RawObject][p as any];
-        const propDef = objectDef.properties[p as any];
-        if (propDef) {
+function createSpecialProperty(
+  client: MinimalClient,
+  objectDef: FetchedObjectTypeDefinition,
+  rawObject: Osdk.Instance<any>,
+  p: keyof typeof rawObject & string | symbol,
+) {
+  const rawValue = rawObject[p as any];
+  const propDef = objectDef.properties[p as any];
+  if (process.env.NODE_ENV !== "production") {
+    invariant(propDef != null && specialPropertyTypes.has(propDef.type));
+  }
+  {
+    {
+      {
+        {
```

Comment:
These just keep the diff simple. We can remove them before merge or in follow up